### ISSUE TITLE
woof-CE needs a new contributor guide and manual

### ISSUE THREAD

**Issue by dimkr** [puppylinux-woof-CE/woof-CE#2068]
https://forum.puppylinux.com/viewtopic.php?f=40&t=2298&start=10

**Comment by dimkr** [comment]
https://gitpod.io/#https://github.com/puppylinux-woof-CE/woof-CE is nice! A pre-configured development environment with predefined "tasks" for stuff like 1download would make it very easy to contribute. Maybe something like fakeroot is needed to run woof-CE or parts of it on the Gitpod machine.

**Comment by CollaboratorGCM** [comment]
I saw a discussion on the forum with a statement to this post. 

Aren't there a tremendous wealth of documentation and videos on GITHUB+GIT+how to setup, use (pull/push), and other such related to projects? If so, can we just provide pointers or search elements that lead persons to that material vs a need to re-create the wheel? 

Or, do we want to look for some automation, like Eclipse (which is excellent), that makes it easy to access and manage things within the project for newbies?

In any decision that is arrived at, there is a support component that must be taken into account. If we use something like an eClipse, there are millions who answer questions for both new&old users. Thus a support requirement is diminished, except for a directional page to get users started.

**Just a question** that might be helpful in "decision options" to satisfy that forum discussion for user(s) who view it.

**Comment by dimkr** [comment]
I agree there's documentation out there, but the number of tools you need to be familiar with (git, shell scripting, GitHub pull requests, ...) might be intimidating to some.

I think a tutorial that instructs people how to open a Gitpod environment, fork woof-CE, do their changes, commit then and open a pull request, would be amazing! Zero setup on the user's machine, the editor is very similar to VS Code (the most popular IDE out there, sorry Eclipse), and everything is done using GUI without using git and the command-line.

For example:
- to open an IDE with woof-CE's code: https://gitpod.io/#https://github.com/puppylinux-woof-CE/woof-CE
- to fork woof-CE: https://www.gitpod.io/docs/fork-a-repo/

**Comment by 01micko** [comment]
I pinned this issue just now.

**Comment by s243a** [comment]
> I saw a discussion on the forum with a statement to this post.
> 
> Aren't there a tremendous wealth of documentation and videos on GITHUB+GIT+how to setup, use (pull/push), and other such related to projects? If so, can we just provide pointers or search elements that lead persons to that material vs a need to re-create the wheel?
> 
> Or, do we want to look for some automation, like Eclipse (which is excellent), that makes it easy to access and manage things within the project for newbies?
> 
> In any decision that is arrived at, there is a support component that must be taken into account. If we use something like an eClipse, there are millions who answer questions for both new&old users. Thus a support requirement is diminished, except for a directional page to get users started.
> 
> **Just a question** that might be helpful in "decision options" to satisfy that forum discussion for user(s) who view it.

I don't use eClipse. It runs too slow on my older machine. I suspect many puppy users are in the same boat.

**Comment by s243a** [comment]
Here's a starting point for anyone that updates github via ssh:

https://docs.github.com/en/github/authenticating-to-github/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent

**Comment by dimkr** [comment]
As a relatively "lightweight" alternative to VS Code that supports its extensions ecosystem and feels familiar, I use https://github.com/dimkr/codile (shameless advertising of my own project). It's just a thin wrapper around https://github.com/eclipse-theia/theia with packaging and CI/CD.

But still, Gitpod is easier because you don't need to install anything on your machine - all you need is a browser. The knowledge from a manual that mentions Gitpod is transferable to VS Code, the most popular IDE out there.

**Comment by CollaboratorGCM** [comment]
I am not sure if this would be of benefit, but if newer ISOs would be released such that a user could/would use the GIT Clone option for ISO download (notwithstanding Distrowatch and forum) be a starting point for users to begin a comfort level?

Not sure, but just a thought.

**Comment by 01micko** [comment]
We need to make an effort to fix up the woof-ce github wiki. 
Things to consider.. many puppy specific
- boot codes
- how to install (frugalpup, old style, etc)
- update the kernel-kit blurb
- update all the other stuff and remove some of it
- how to configure a woof build

The list could go on.

**Comment by sc0ttj** [comment]
I've banged on about the need for a proper Developer Handbook, lots..  Hopefully it'll go somewhere..

The CRUX Handbooks are simple, and really nice - using lots of small, easy code examples that users can copy & paste - so much easier than trying to document a bunch of quirky/archaic Gtkdialog GUIs..  

In term of being any kind of helpful "documentation", the forum is a mess - and worse, even the most regular users there don't seem to understand development happens here, and they are largely downstream - as the latest huge (and excellent looking) PR from dimkr shows..  

And I still think a "definition of done" for official releases if a sensible thing to have, and gave my own ideas about what that could be now, and in the future in some other Issue.

**Comment by dimkr** [comment]
Started working on documentation: https://github.com/puppylinux-woof-CE/woof-CE/wiki/Building-a-Puppy-on-GitHub

The plan: this + a guide that instructs people how to fork woof-CE, change something, commit the changes, push, publish a release in their fork (+ tips what and how to test) and open a pull request to woof-CE.

**Comment by SamuelBanya** [comment]
Hey there,

I really think Puppy Linux is awesome after going through the gambit of different Linux distros in the past couple of years.

I have wanted to contribute in some shape or form, but am kind of lost as well as to "What" is needed to be done, as well as the "How" to actually do it.

I was thinking to help out with the development side of things to learn more programming skills, as well as visual assets since I enjoy doing artwork on the side as well.

Can anyone point me in the right direction on this, as I wasn't too sure on how to proceed after reading the related page on the topic:
https://puppylinux.com/team.html

If anyone could point me to any 'Beginner' tier issues, as well as 'how' to actually make the changes via GitHub commits, and also what artwork is needed, that would be cool too :)

Thanks,

Sam

**Comment by SamuelBanya** [comment]
Hey @dimkr 
Any update on this?

Would love to help out, but have zero clue on how to start to be honest.

**Comment by dimkr** [comment]
> Hey @dimkr Any update on this?

Nope, not yet.

**Comment by dimkr** [comment]
https://github.com/puppylinux-woof-CE/woof-CE/wiki/woof-CE-Contributor-101

Comments are welcome.
